packages/store/src/storage.ts

```diff
@@ -1044,6 +1046,8 @@ class Storage {
     // at this point document is either created or existed before
     const [firstAttachmentKey] = Object.keys(_attachments);
     const buffer = this.getBufferManifest(body._attachments[firstAttachmentKey].data as string);
+    const readable = Readable.from(buffer);
+    const tarballStats = await this.getTarballStats(versions[versionToPublish], readable);
```

Comment:
I'm concerned this might slow down things at least the first time a tarball is requested, what's are your thoughts?

Reply:
It's only taking time if `dist` isn't filled. Packages from npm uplink for example are not impacted at all. When publishing to verdaccio directly, then it does take some time, but the npm registry does it the same way. I don't think it will make a significant difference to users.